src/plugins/intel_npu/src/plugin/npuw/compiled_model.cpp

```diff
@@ -526,6 +530,49 @@ ov::npuw::CompiledModel::CompiledModel(const std::shared_ptr<ov::Model>& model,
     LOG_DEBUG("CompiledModel is being deserialized, skipping the full constructor flow...");
 }
 
+void ov::npuw::CompiledModel::identify_host_gather_property(const std::shared_ptr<ov::Model>& model) {
+    // Identify based on the config
+    try {
+        auto compiler_version = get_plugin()->get_core()->get_property("NPU", ov::intel_npu::compiler_version);
+        if (compiler_version >= ONEAPI_MAKE_VERSION(7, 21)) {
+            // Force quantized host gather
+            LOG_INFO("Forcing NPUW_HOST_GATHER_QUANT property.");
+            std::map<std::string, std::string> host_gather_cfg;
+            host_gather_cfg["NPUW_HOST_GATHER_QUANT"] = "YES";
+            host_gather_cfg["NPUW_HOST_GATHER"] = "NO";
+            m_cfg.update(host_gather_cfg);
+        } else {
+            if (m_cfg.get<::intel_npu::NPUW_HOST_GATHER_QUANT>()) {
+                NPUW_ASSERT(false && "NPU compiler doesn't support tail quantization! "
+                                     "Please either use NPUW_HOST_GATHER instead of NPUW_HOST_GATHER_QUANT "
+                                     "or get the latest driver.");
+            }
+        }  // compiler_version
+    } catch (...) {
+        // No device, keeping properties as is. Just do a sanity check
+        if (m_cfg.get<::intel_npu::NPUW_HOST_GATHER_QUANT>() && m_cfg.get<::intel_npu::NPUW_HOST_GATHER>()) {
+            NPUW_ASSERT(false && "Conflicting configuration: NPUW_HOST_GATHER and NPUW_HOST_GATHER_QUANT can't be "
+                                 "enabled together!");
+        }
+    }
+
+    // Verify NPUW_HOST_GATHER_QUANT based on the patterns (for tail vocab)
+    if (m_cfg.get<::intel_npu::NPUW_HOST_GATHER_QUANT>()) {
+        ov::npuw::patterns::opt::Context ctx;
+        ov::pass::GraphRewrite rewr;
+        rewr.add_matcher<ov::npuw::patterns::opt::HostGatherQuantAsymm>(std::ref(ctx), true);
+        rewr.run_on_model(model);
+
+        if (!ctx.found_host_gather_quant()) {
+            LOG_INFO("Couldn't match NPUW_HOST_GATHER_QUANT-related patterns. Enabling NPUW_HOST_GATHER instead.");
+            std::map<std::string, std::string> host_gather_cfg;
+            host_gather_cfg["NPUW_HOST_GATHER_QUANT"] = "NO";
+            host_gather_cfg["NPUW_HOST_GATHER"] = "YES";
+            m_cfg.update(host_gather_cfg);
+        }
+    }
```

Comment:
for automatic decision making, I believe it should be smt like this:

```
if (supports_asym_vocab && found_host_gather_quant) {
    host_gather_cfg["NPUW_HOST_GATHER_QUANT"] = "YES";
    host_gather_cfg["NPUW_HOST_GATHER"] = "NO";
} else {
    host_gather_cfg["NPUW_HOST_GATHER_QUANT"] = "NO";
    host_gather_cfg["NPUW_HOST_GATHER"] = "YES";
}
```
but again, this essentially discards the user preference. Imagine we're running subgraph dumping job on Linux, how would we force the right configuration where there's no device at all?

Reply:
Done

Reply:
For Linux the config will be preserved as specified by the user